src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -1,14 +1,55 @@
 const { readFile } = require('fs/promises');
 const { join } = require('path');
 const { cwd } = require('process');
+const { getPackument } = require('query-registry');
 
-async function getDependencies() {
-  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+const getDependencies = (function () {
+  let dependencies = null;
 
-  // The order of the alternatives is important!
-  // The regex engine will favor the first pattern on an alternation
-  // even if the other alternatives are subpatterns
-  return dependencies.split(/\r\n|\n|\r/).filter((line) => line !== '');
+  return async () => {
+    if (!dependencies) {
+      dependencies = {};
+      const dependencyList = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+      dependencyList
+        .split(/\r\n?|\n/g)
+        .filter((line) => !!line)
```

Comment:
Interesting operator `!!`

Reply:
Another way people do this is `.filter(line => Boolean(line))` or even `.filter(Boolean)`